### DISCUSSION TITLE
whereis defaultSettings.json?

### DISCUSSION THREAD

**Discussion by alanhe421** [coder/code-server#5747] [General] [answered]
404 found in path
![image](https://user-images.githubusercontent.com/9245110/199523195-c9f2403b-12a5-46c1-a171-5bd573c17718.png)

I want to edit the file by shell.

**Comment by jsjoeio** [comment, marked as answer]
I'm actually not sure 🤔 Usually I open it via the Command Palette and it shows the path. But I'm not seeing it:

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/199595184-abd42ca1-af97-4db3-b6f9-25396751ae66.png">

The `settings.json` for the User are stored here on my macOS though in case that helps:
`/Users/<me>/.local/share/code-server/User/settings.json`

**Reply by alanhe421** [reply]
ok.

now I have to use `/Users/<me>/.local/share/code-server/User/settings.json`.


and I have to see source code if want to know how to control defaultSettings.
